### DISCUSSION TITLE
[Feature proposal]: ":fullscreen" CSS pseudo-class

### DISCUSSION THREAD

**Discussion by MasseGuillaume** [tailwindlabs/tailwindcss#1578] [Ideas]
I would love to see support for https://developer.mozilla.org/en-US/docs/Web/CSS/:fullscreen

**Comment by pinkplus** [comment]
+1

In the meanwhile I can achieve it by adding a custom plugin in tailwind.config.js:

```javascript
  plugins: [
    plugin(function({ addVariant }) {
      addVariant('fullscreen', '&:fullscreen');
    })
  ],
```

**Reply by demccormack** [reply]
Thanks a lot, this is great! In my Vite project, I needed to omit the plugin() function call, so the plugins section of my tailwind.config.js is
```
  plugins: [
    function ({ addVariant }) {
      addVariant('fullscreen', '&:fullscreen');
    },
```

**Comment by unrevised6419** [comment]
If someone needs support also for `group` and `peer`

```ts
plugins: [
	plugin(({ addVariant }) => {
		addVariant('fullscreen', '&:fullscreen');
		addVariant('group-fullscreen', ':merge(.group):fullscreen &');
		addVariant('peer-fullscreen', ':merge(.peer):fullscreen ~ &');
	}),
],
```

**Comment by karloskar** [comment]
When using v4 the same thing can be accomplished by adding `@variant` statements.

```css
@variant fullscreen (&:fullscreen);
@variant group-fullscreen (.group:fullscreen &);
@variant peer-fullscreen (.peer:fullscreen &);
```

**Reply by unrevised6419** [reply]
Shouldn't there be a tilda (`~`) between `.peer:fullscreen` and `&`? 🤔 

```css
@variant peer-fullscreen (.peer:fullscreen ~ &);
```

**Reply by RobinMalfait** [reply]
> When using v4 the same thing can be accomplished by adding `@variant` statements.
> 
> 
> 
> ```css
> 
> @variant fullscreen (&:fullscreen);
> 
> @variant group-fullscreen (.group:fullscreen &);
> 
> @variant peer-fullscreen (.peer:fullscreen &);
> 
> ```

Drop the `group-fullscreen` and `peer-fullscreen` variants, they are automatically available 👍

E.g.: https://play.tailwindcss.com/P93mEXuJr2

![image](https://github.com/user-attachments/assets/4ea97b9e-bc0d-4b07-a557-332372bdd4f0)

**Reply by karloskar** [reply]
You're right @RobinMalfait! No need for the group- and peer- versions.

Also a nice and very useful use of https://play.tailwind.com :pray:
